tests/test_series.py

```diff
@@ -1192,3 +1192,20 @@ def test_types_to_numpy() -> None:
     check(assert_type(s.to_numpy(), np.ndarray), np.ndarray)
     check(assert_type(s.to_numpy(dtype="str", copy=True), np.ndarray), np.ndarray)
     check(assert_type(s.to_numpy(na_value=0), np.ndarray), np.ndarray)
+
+
+def test_where() -> None:
+    s = pd.Series([1, 2, 3], dtype=int)
+
+    def cond1(x: int) -> bool:
+        return x % 2 == 0
+
+    check(assert_type(s.where(cond1), pd.Series[int]), pd.Series, int)
```

Comment:
Because `pd.Series[int]` is only known to the type checker, and not to the runtime, you need to surround that type in quotes, e.g.:
```python
    check(assert_type(s.where(cond1), "pd.Series[int]"), pd.Series, int)
```

Reply:
Amended previous commit. I was previously only running the `mypy` poetry workflow and so missed the actual test failures. I now confirmed everything is passing under `test`.